setup.py

```diff
@@ -52,26 +54,44 @@
 )
 
 
-py_version = sys.version[:3]
+py_version = sys.version_info
 
 SCRIPT1 = 'nose2'
-SCRIPT2 = 'nose2-%s' % (py_version,)
+SCRIPT2 = 'nose2-%s.%s' % (py_version.major, py_version.minor)
+
+
+def parse_requirements(requirement_file):
+    requirements = []
+    with open(requirement_file) as file_pointer:
+        for line in file_pointer:
+            if line.strip() and not line.strip().startswith('#'):
+                requirements.append(line.strip())
+    return requirements
+
+
+def per_version_requirements(extra_requires_dict):
```

Comment:
Am I missing something? Is this getting called anywhere? If not, how is travis passing? <confused>

Also, you don't need to pass in the old version, you can just have something like
```
params['extra_require'].update(per_version_requirements())
```

Reply:
you don't miss anything.

Reply:
fixed

Reply:
Weird that it was passing though.

Reply:
In fact, tox is not used so the build is passing but tox is out of service.
I commit once the full tox run is over.

Reply:
yes, but your travis should still be running this code when you do `pip install .` Then the tests should fail if proper requirements are not installed.

(Wouldn't fail with the old `.travis.yaml`, this should have failed with the new one)